### ISSUE TITLE
Crush after attempt to move an amount of dehydrated meat to inventory in russian localization

### ISSUE THREAD

**Issue by stasalcatraz** [CleverRaven/Cataclysm-DDA#13509]
I'm trying to move some dehydrated meat from fridge to inventory via advanced inventory management. When i press "m" to move amount of items, game crushes with message

terminate called after throwing an instance of 'std::length_error'
  what():  basic_string::_S_create

Game version e10c546 tiled for linux with russian language in options. In english anything ok.

**Comment by stasalcatraz** [comment]
The same with hydrogen.

**Comment by drbig** [comment]
Can't replicate here at 88a4218. The error message you got would indicate something screwy going on with strings, so the locale might be a factor here... This being AIM, still, сушёное мясо doesn't break anything here.

**Comment by Night-Pryanik** [comment]
Cannot confirm it on 0.C-3662 with fridge and vehicle's mini-fridge. Win 10, though.

**Comment by stasalcatraz** [comment]
I had to give my computer with ubuntu to service center, but I was able to transfer my saves to another computer with Windows 8. I install version 2d80646 and I've got crush again. Here is my save with this issue https://www.dropbox.com/s/e4p78kngg39em2x/inventoryError.tgz?dl=0

**Comment by VlasovVitaly** [comment]
Was able to reproduce. Trying to fix that.

**Comment by VlasovVitaly** [comment]
Pretty funny bug not related to this save game and item name. It's about item quantity.
Exception is raised in 
```
std::string string_input_win()
...
mvwprintw( w, starty, startx, std::string( scrmax, ' ' ).c_str() );
...
```
scrmax = -1 in this call and everything is crush.
Error in input string length. When it is too big (item quantity is > 10) scrmax will be -1 and everything will crush.
Screen with quantity < 10:
![right_length](https://cloud.githubusercontent.com/assets/5244047/9758340/a77b5db2-56ee-11e5-899f-9c7590a96d9d.png)

Input field is almost on edge.

I will fix Russian input string and bug will come away.
But generally,  string_input_win() should be fixed and i know how. If length of input string is too big then it must start newline for numbers input.

**Comment by VlasovVitaly** [comment]
Btw this message should be changed. It's not clean for understanding.

**Comment by DanmakuDan** [comment]
I think the message for <destination doesn't have enough space left> and <destination can hold all of the movable item> are switched, moving something that is not reaching a limit says 'Destination can only hold X! Move how many?'. Moving something that would overload the limit says 'How many do you want to move?'. The messages make more sense for the opposite situation.

**Comment by VlasovVitaly** [comment]
If destination can hold all amount of item we actually don't care about this. We need know storage limit only when destination can't hold all amount.
So, my suggestion is:
if (destitation_storage_limit >= item_amoun) {
 "How many to move? [has %d] (0 to cancel)
} else {
 // Assume that you have more items that destination can hold.
 "Destination can hold only %d. How many to move? (0 to cancel)
}

**Comment by Night-Pryanik** [comment]
Should be closed after #13528?
